omegaconf/omegaconf.py

```diff
@@ -425,11 +428,25 @@ def register_new_resolver(
             ${foo:${bar}} will always return the same value regardless of the value of
             `bar` if the cache is enabled for `foo`.
         """
-        assert callable(resolver), "resolver must be callable"
-        # noinspection PyProtectedMember
-        assert (
-            name not in BaseContainer._resolvers
-        ), "resolver {} is already registered".format(name)
+        if not callable(resolver):
+            raise TypeError("resolver must be callable")
+        if not name:
+            raise ValueError("cannot use an empty resolver name")
+
+        existing = OmegaConf._get_resolver(name)
+        if existing is not None:
+            # DEPRECATED: remove this check in 2.2
+            # (it is only there because some resolvers may be registered with
+            # the legacy function and thus would not be `partial` objects)
+            # Also the typing may be updated accordingly.
```

Comment:
an alternative here is to change the legacy resolver to use a trivial wrapper as well.

Reply:
Do you feel strongly about it? If not I suggest we leave it as is, I'll take care of the clean-up for 2.2.

Reply:
This has been reverted, new version is in 3554985